### PR TITLE
[Backend] As a system, I can have the capacity to store scrapped result links

### PR DESCRIPTION
**Pull request by mosharaf13**
Closes https://github.com/mosharaf13/google-scrapper-ruby/issues/40 

## What happened 👀

- Created table to store result links from google search parsed data
- Returned stored data from controller instead of dummy data

## Insight 📝

- I was returning dummy result links from search stat show method. Now it returns actual data from db.

## Proof Of Work 📹

Screen Shot 2023-06-01 at 6 06 41 PM

### CODE REVIEW

File: app/models/result_link.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class ResultLink < ApplicationRecord
+  enum link_type: { ads_top: 0, ads_page: 1, non_ads: 2 }
```
[the remark this answers is not in the document]
Comment (mosharaf13) [reply app/models/result_link.rb:4]: I was thinking about keeping count for all the ads in a page. Although, it is unnecessary. Removing.

[the remark this answers is not in the document]
Comment (mosharaf13) [reply app/models/result_link.rb]: Fixed in ad8a1a8163a030b7a14ce0b33a76f1a4c21f5b4c

File: db/migrate/20230601085414_create_result_links.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class CreateResultLinks < ActiveRecord::Migration[7.0]
+  def change
+    enable_extension 'citext' unless extension_enabled?('citext')
+
+    create_table :result_links do |t|
+      t.references :search_stat, null: false, foreign_key: true
+      t.integer :link_type, null: false
+      t.citext :url, null: false
+
+      t.timestamps default: -> { 'CURRENT_TIMESTAMP' }
```
Comment (sanG-github) [inline db/migrate/20230601085414_create_result_links.rb:10]: I think it's enough, did you face any problems with it? 🤔 
```suggestion
      t.timestamps
```

**Review by sanG-github — APPROVED** [review]

File: db/migrate/20230601085414_create_result_links.rb (hunk 2 above)
Comment (mosharaf13) [reply db/migrate/20230601085414_create_result_links.rb]: Umm, no. Took idea from another codebase. Thought that's how timestamps should be initialized in rails.

File: db/seeds.rb [hunk 3]
```diff
@@ -9,6 +9,7 @@
 require 'fabrication'
 
 # Generate dummy data for SearchStat
-100.times do
-  Fabricate(:search_stat)
+10.times do
+  user = User.where(email: 'user@demo.com').first_or_create(Fabricate.attributes_for(:user, email: 'user@demo.com'))
+  Fabricate.times(100, :search_stat_parsed_with_links, user: user)
 end
```
[the remark this answers is not in the document]
Comment (mosharaf13) [reply db/seeds.rb]: Ahha. What was I thinking 😬. Compiler are smart though 😉 .

**Review by longnd — APPROVED** [review]

File: app/views/search_stats/show.html.erb [hunk 4]
```diff
@@ -63,9 +63,9 @@
     <div class="card-body">
       <h5 class="card-title">List of URLs of AdWords Advertisers in Top Position</h5>
       <ul class="list-group">
-        <% @search_stat.adwords_urls.each do |url| %>
+        <% @search_stat.result_links.ads_top.or(@search_stat.result_links.ads_page).each do |result_link| %>
           <li class="list-group-item">
-            <%= link_to url, url, target: "_blank" %>
+            <%= link_to result_link.url, result_link.url, target: "_blank" %>
```
Comment (sanG-github) [inline app/views/search_stats/show.html.erb]: Is there any reason we need to open it in a new window? 🤔 
```suggestion
            <%= link_to result_link.url, result_link.url %>
```

File: db/migrate/20230601085414_create_result_links.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+class CreateResultLinks < ActiveRecord::Migration[7.0]
+  def change
+    enable_extension 'citext' unless extension_enabled?('citext')
+
+    create_table :result_links do |t|
+      t.references :search_stat, null: false, foreign_key: true
+      t.integer :link_type, null: false
+      t.citext :url, null: false
```
[the remark this answers is not in the document]
Comment (mosharaf13) [reply db/migrate/20230601085414_create_result_links.rb:8]: Fixed in 3613756ca31e9175e56145463256782eca10d535

**Pull request merged**